### PR TITLE
Add workflow for ITS PID study

### PR DESCRIPTION
**Pull request by fmazzasc**
@mconcas

### CODE REVIEW

**Review by mconcas — DISMISSED** [review]

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    auto& ITSTPCtrack = ITSTPCtracks[iTrack];
+    if (ITSTPCtrack.getRefITS().getSource() == GTrackID::ITSAB) // excluding Afterburned tracks
+      continue;
+
+    particle part;
+    auto ITStrack = recoData.getITSTrack(ITSTPCtrack.getRefITS());
+    auto TPCtrack = recoData.getTPCTrack(ITSTPCtrack.getRefTPC());
+
+    part.sign = ITSTPCtrack.getSign();
+    part.clSizesITS = getTrackClusterSizes(ITStrack);
```
Comment (stefanopolitano) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:223]: I am not 100% sure, but, if I remember correctly, at the time we performed the study we used to scale the cluster size (or at least the average cluster size) by the cosine of the lambda angle, being lambda the angle between the track and the ITS layer to account for the inclination of the track. Do you plan to do it locally maybe? Like the computation of the average cluster size?

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    part.clSizeL2 = part.clSizesITS[2];
+    part.clSizeL3 = part.clSizesITS[3];
+    part.clSizeL4 = part.clSizesITS[4];
+    part.clSizeL5 = part.clSizesITS[5];
+    part.clSizeL6 = part.clSizesITS[6];
+
+    // PID info
+    part.dEdx = TPCtrack.getdEdx().dEdxTotTPC;
+    part.nClusTPC = TPCtrack.getNClusters();
+    // 7% resolution for all particles
```
Comment (stefanopolitano) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:246]: Just a curiosity from my side, where did you get this value? I remember we already used it at the time of the pilot beam, but I can not recall where we took it from

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+using TrackITS = o2::its::TrackITS;
+using TrackTPC = o2::tpc::TrackTPC;
+using TrackITSTPC = o2::dataformats::TrackTPCITS;
+using PIDResponse = o2::tpc::PIDResponse;
+using DCA = o2::dataformats::DCA;
+using PID = o2::track::PID;
+
+// structure for storing the output tree
+struct particle {
+  int pdg = -1;
```
Comment (stefanopolitano) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:76]: If I am not mistaken, this information is never updated even when the `useMC` flag is enabled

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+  int sign = -1;
+  float rofBC;
```
Comment (stefanopolitano) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx]: If I am not mistaken, this information is never updated

**Review by stefanopolitano — COMMENTED** [review]
Hi @fmazzasc ! Thanks a lot for this important contribution ;) . I just have few minor comments

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx (hunk 1 above)
Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:223]: We can do it offline and save disk space

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx (hunk 2 above)
Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:246]: It's the average tpc resolution of the 2023 sample

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx (hunk 3 above)
Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx:76]: fixed, MC info added

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx (hunk 4 above)
Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx]: rofBC removed, not needed anymore

**Review by stefanopolitano — APPROVED** [review]

File: Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+    LOGP(debug, "Found {} ITS labels.", mcLabelsITS.size());
+    LOGP(debug, "Found {} TPC labels.", mcLabelsTPC.size());
+  }
+
+  for (unsigned int iTrack{0}; iTrack < ITSTPCtracks.size(); ++iTrack) {
+
+    auto& ITSTPCtrack = ITSTPCtracks[iTrack];
+    if (ITSTPCtrack.getRefITS().getSource() == GTrackID::ITSAB) // excluding Afterburned tracks
+      continue;
+
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/src/PIDStudy.cxx]:
```suggestion
    if (ITSTPCtrack.getRefITS().getSource() == GTrackID::ITSAB) { // excluding Afterburned tracks
      continue;
    }

```

**Review by mconcas — APPROVED** [review]

**Pull request merged**